### PR TITLE
slt: add slt file demoing session windows

### PR DESCRIPTION
**Pull request by sploiselle**
Requested on [Slack](https://materializeinc.slack.com/archives/CNVRXGFDJ/p1679314661753049?thread_ts=1679077268.922059&cid=CNVRXGFDJ): porting over [this code](https://github.com/MaterializeInc/materialize/issues/8698#issuecomment-1474168849) to `slt`.

### Motivation

This PR adds a known-desirable feature.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: test/sqllogictest/session-window-wmr.slt [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+    event_time timestamp,
+    -- who?
+    user_id int,
+    -- how much?
+    worth decimal
+);
+
+statement ok
+CREATE MATERIALIZED VIEW event_session AS
+	WITH MUTUALLY RECURSIVE
```
Comment (aalexandrov) [inline test/sqllogictest/session-window-wmr.slt:24]: nit: can we replace the tabs here with 4 spaces?

File: test/sqllogictest/session-window-wmr.slt [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+SELECT * FROM event_session
+----
+1
+["2021-01-01 10:00:00","2021-01-01 10:11:00")
+1
+["2021-01-01 10:13:00","2021-01-01 10:26:00")
+2
+["2021-01-01 10:02:00","2021-01-01 10:09:00")
+
+query TTTTT
```
Comment (aalexandrov) [inline test/sqllogictest/session-window-wmr.slt:350]: Is there a way to render each row of the expected result in a single line?

**Review by aalexandrov — APPROVED** [review]
Thanks for contributing this.

Two minor nits, otherwise LGTM.

File: test/sqllogictest/session-window-wmr.slt (hunk 2 above)
Comment (sploiselle) [reply test/sqllogictest/session-window-wmr.slt]: slt doesn't support this but we can fake it in SQL

**Pull request merged**